imports/plugins/core/graphql/server/no-meteor/xforms/cart.js

```diff
@@ -56,8 +56,17 @@ async function xformCartItem(context, catalogItems, products, cartItem) {
     throw new ReactionError("invalid-param", `Product with ID ${productId} has no variant with ID ${variantId}`);
   }
 
+  const variantSourceProduct = products.find((product) => product._id === variantId);
+  const catalogVariantId = variantSourceProduct.ancestors[1];
+  const catalogVariant = catalogProduct.variants.find((cVariant) => cVariant._id === catalogVariantId);
```

Comment:
`ancestors[1]` is not correct because sometimes there's only one item in the `ancestors` array. `variantId` here may be referring to a variant or option ID, and we've already found `variant` (which may be an option) with the `findVariantInCatalogProduct` call above.

So I think removing everything above here and changing `catalogVariant` to `variant` below will make this correct.

This would be a good time to add a unit test to prove this is working, too.